File: mycollab-mobile/src/main/java/com/esofthead/mycollab/mobile/module/crm/view/activity/ActivityTaskUrlResolver.java
package com.esofthead.mycollab.mobile.module.crm.view.activity;

import com.esofthead.mycollab.common.UrlEncodeDecoder;
import com.esofthead.mycollab.eventmanager.EventBusFactory;
import com.esofthead.mycollab.mobile.module.crm.CrmUrlResolver;
import com.esofthead.mycollab.mobile.module.crm.events.ActivityEvent;
import com.esofthead.mycollab.module.crm.domain.Task;

/**
 * 
 * @author MyCollab Ltd.
 * @since 4.4.0
 * 
 */
public class ActivityTaskUrlResolver extends CrmUrlResolver {
	public ActivityTaskUrlResolver() {
		this.addSubResolver("add", new TaskAddUrlResolver());
		this.addSubResolver("edit", new TaskEditUrlResolver());
		this.addSubResolver("preview", new TaskPreviewUrlResolver());
	}

	public static class TaskAddUrlResolver extends CrmUrlResolver {
		@Override
		protected void handlePage(String... params) {
			EventBusFactory.getInstance().post(
					new ActivityEvent.TaskAdd(this, new Task()));
		}
	}

	public static class TaskEditUrlResolver extends CrmUrlResolver {
		@Override
		protected void handlePage(String... params) {
			String decodeUrl = UrlEncodeDecoder.decode(params[0]);
			int meetingId = Integer.parseInt(decodeUrl);
			EventBusFactory.getInstance().post(
					new ActivityEvent.TaskEdit(this, meetingId));
		}
	}

	public static class TaskPreviewUrlResolver extends CrmUrlResolver {
		@Override
		protected void handlePage(String... params) {
			String decodeUrl = UrlEncodeDecoder.decode(params[0]);
			int accountId = Integer.parseInt(decodeUrl);
			EventBusFactory.getInstance().post(
					new ActivityEvent.TaskRead(this, accountId));
		}
	}
}

File: mycollab-mobile/src/main/scala/com/mycollab/mobile/module/crm/view/activity/ActivityTaskUrlResolver.scala
package com.mycollab.mobile.module.crm.view.activity

import com.mycollab.common.UrlTokenizer
import com.mycollab.eventmanager.EventBusFactory
import com.mycollab.mobile.module.crm.CrmUrlResolver
import com.mycollab.mobile.module.crm.events.ActivityEvent
import com.mycollab.module.crm.domain.CrmTask

/**
  * @author MyCollab Ltd
  * @since 5.0.9
  */
class ActivityTaskUrlResolver extends CrmUrlResolver {
  this.addSubResolver("add", new TaskAddUrlResolver)
  this.addSubResolver("edit", new TaskEditUrlResolver)
  this.addSubResolver("preview", new TaskPreviewUrlResolver)
  
  class TaskAddUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      EventBusFactory.getInstance().post(new ActivityEvent.TaskAdd(this, new CrmTask))
    }
  }
  
  class TaskEditUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      val meetingId = UrlTokenizer(params(0)).getInt
      EventBusFactory.getInstance().post(new ActivityEvent.TaskEdit(this, meetingId))
    }
  }
  
  class TaskPreviewUrlResolver extends CrmUrlResolver {
    protected override def handlePage(params: String*) {
      val accountId = UrlTokenizer(params(0)).getInt
      EventBusFactory.getInstance().post(new ActivityEvent.TaskRead(this, accountId))
    }
  }
  
}
